storage/s3.go

```diff
@@ -62,29 +62,43 @@ type S3 struct {
 	partSize          int64
 }
 
-type s3Opt func(s3 *S3)
+type S3Opt func(s3 *S3) error
```

Comment:
Question: what is the reason for exporting this signature?

IMO it should stay private in order to only provide `With*` functions here. Or am I missing use cases?